src/lib/features/feature-toggle/tests/feature-toggle-service.limit.test.ts

```diff
@@ -14,71 +15,145 @@ const alwaysOnFlagResolver = {
     },
 } as unknown as IFlagResolver;
 
-test('Should not allow to exceed strategy limit', async () => {
-    const LIMIT = 3;
-    const { featureToggleService, featureToggleStore } =
-        createFakeFeatureToggleService({
-            getLogger,
-            flagResolver: alwaysOnFlagResolver,
-            resourceLimits: {
-                featureEnvironmentStrategies: LIMIT,
-            },
-        } as unknown as IUnleashConfig);
-
-    const addStrategy = () =>
-        featureToggleService.unprotectedCreateStrategy(
-            { name: 'default', featureName: 'feature' } as IStrategyConfig,
-            { projectId: 'default', featureName: 'feature' } as any,
-            {} as IAuditUser,
+describe('Strategy limits', () => {
```

Comment:
I moved strategy and constraint testing into a shared describe block because they both use strategy methods on the service. This keeps them separate from the flags testing block. We can split this out into two files if we think that's better, but this seems sensible to me for now.